### PR TITLE
Add no-intro option (Issue #146)

### PR DESCRIPTION
**Pull request by agxies**
I added the functionality to not show the intro message when the program is started (#146).
Can be activated via `--no-intro` flag or the config file.

### CODE REVIEW

**Comment by cjmayo** [comment]
The concept of intro as an item, and something that is configurable as one of "parts" setting in linkcheckerrc, already exists in the code. A `--no-intro` that still runs write_intro() doesn't seem very clean to me. And having two similar settings in linkcheckerrc even less desirable. Maybe having just a command line `--no-intro` to remove intro from the parts argument before creating the logger would be OK. Not sure how cleanly that could be done.

**Comment by agxies** [comment]
Sorry, I totally overlooked this option in the config file. I now have changed the implementation so that the command line option uses the parts setting. I'm just not quite sure if it is clean enough like that.

File: linkchecker [hunk 1]
```diff
@@ -669,6 +675,11 @@ if options.profile:
     do_profile = options.profile
 if options.quiet:
     config["logger"] = config.logger_new("none")
+if options.no_intro:
+    if not config["text"] or config["text"]["parts"][0] == "all":
+        config["text"]["parts"] = ["realurl", "result", "extern", "base", "name", "parenturl", "info", "warning", "dltime", "checktime", "url", "outro", "stats"]
```
Comment (cjmayo) [inline linkchecker:680]: I can see why you did this but hardcoding the list of parts seems too high a price to pay.

**Comment by cjmayo** [comment]
I can't see a way to do this without adding complexity. Because the current assumption is everything is enabled unless there is an explicit list of parts.

I didn't realise that configuration option was there either before this - so I learnt something!  Probably someone who doesn't want the intro would be happy to set the option and leave it.

**Comment by agxies** [comment]
Learning something is always great :)
Another option may be to change the default behavior. So that by selecting 'all', the parts object will automatically get the list of all parts, but that won't solve the complexity issue either.

From my perspective the complete implementation of the parts function feels somehow inconvenient. Because to deactivate one of the parts, it is mandatory to add all others in the config file. Maybe another approach with simple boolean variables would be sensible, so that every feature can be enabled/disabled individually. This would make it also much easier to have flags for the command line. Don't know what your opinion would be on that?

**Comment by cjmayo** [comment]
Closing because I can't imagine this can be done. Maybe I lack imagination...

**Pull request closed without merging**